File: src/main/resources/static/js/url.js


var Url = {};
Url.PAGE = _contextPath;
if(Url.PAGE.lastIndexOf('/') === 0 && Url.PAGE.length === 1) {
    Url.PAGE = '';
}
Url.API = Url.PAGE + '/api';

// 页面 url
var Page = {};
{
    var page = Url.PAGE ;
    Page.SYSTEM = page + '/system';
    Page.USER = page + '/user';

    {
        system = {};
        system.VIEW_PAGE = Page.SYSTEM + '/view_page';
        system.VIEW_PAGE_CATEGORY = Page.SYSTEM + '/view_page_category';
        system.VIEW_PAGE_API = Page.SYSTEM + '/view_page_api';
        system.VIEW_PAGE_COMPONENT = Page.SYSTEM + '/view_page_component';
        system.VIEW_PAGE_COMPONENT_API = Page.SYSTEM + '/view_page_component_api';

        system.API = Page.SYSTEM + '/api';
        system.API_CATEGORY = Page.SYSTEM + '/api_category';

        system.DICTIONARY = Page.SYSTEM + '/dictionary';
        system.DICTIONARY_CATEGORY = Page.SYSTEM + '/dictionary_category';

        system.MAIL = Page.SYSTEM + '/mail';

        system.OPERATION_LOG = Page.SYSTEM + '/operation_log';

        system.ATTACHMENT = Page.SYSTEM + '/attachment';

        Page.System = system;
    }

    {
        user = {};
        user.LOGIN = Page.USER + '/login';
        user.LOGOUT = Page.USER + '/logout';

        user.ROLE = Page.USER + '/role';
        user.ROLE_AUTHORITY = Page.USER + '/role_authority';

        user.VIEW_MENU = Page.USER + '/view_menu';
        user.VIEW_MENU_CATEGORY = Page.USER + '/view_menu_category';

        user.USER_ROLE = Page.USER + '/user_role';

        user.DEPARTMENT = Page.USER + '/department';

        Page.User = user;
    }

}

// api url
var Api = {};
{
    var api = Url.API;
    Api.SYSTEM = api + '/system';
    Api.USER = api + '/user';

    {
        var system = {};
        system.VIEW_PAGE = Api.SYSTEM + '/view_page';
        system.VIEW_PAGE_CATEGORY = Api.SYSTEM + '/view_page_category';
        system.VIEW_PAGE_API = Api.SYSTEM + '/view_page_api';
        system.VIEW_PAGE_COMPONENT = Api.SYSTEM + '/view_page_component';
        system.VIEW_PAGE_COMPONENT_CATEGORY = Api.SYSTEM + '/view_page_component_category';
        system.VIEW_PAGE_COMPONENT_API = Api.SYSTEM + '/view_page_component_api';

        system.API = Api.SYSTEM + '/api';
        system.API_CATEGORY = Api.SYSTEM + '/api_category';

        system.DICTIONARY = Api.SYSTEM + '/dictionary';
        system.DICTIONARY_CATEGORY = Api.SYSTEM + '/dictionary_category';

        system.MAIL = Api.SYSTEM + '/mail';

        system.OPERATION_LOG = Api.SYSTEM + '/operation_log';

        // system.REQUEST_LOG = Api.SYSTEM + '/operation_log';

        system.ATTACHMENT = Api.SYSTEM + '/attachment';

        Api.System = system;
    }

    {
        var user = {};
        user.LOGIN = Api.USER + '/login';
        user.LOGOUT = Api.USER + '/logout';

        user.ROLE = Api.USER + '/role';
        user.ROLE_AUTHORITY = Api.USER + '/role_authority';

        user.VIEW_MENU = Api.USER + '/view_menu';
        user.VIEW_MENU_CATEGORY = Api.USER + '/view_menu_category';

        user.USER_ROLE = Api.USER + '/user_role';

        user.DEPARTMENT = Api.USER + '/department';

        Api.User = user;
    }

}


// function getDictionaryCode(field, codeSlug){
//     ajax({url:api.system+'/dictionary/getCode/byFieldAndCodeSlug',data:{field:field,codeSlug:codeSlug},success:function (result) {
//             if(result.data){
//                 return result.data;
//             }
//         }})
// }

File: src/main/java/com/gioov/nimrod/common/Url.java
package com.gioov.nimrod.common;

/**
 * @author godcheese [godcheese@outlook.com]
 * @date 2018-02-22
 */
public class Url {

    public static final String PATH_SEPARATOR = "/";
    public static final String ALL_PATH_PATTERN = "/**";

    public static final String API = "/api";

    public static final String API_ALL_PATTERN = API + "/**";

    /**
     * 静态资源 url
     */
    public static final String[] STATIC = {"robots.txt", "/**.ico", "/assets/**", "/css/**", "/js/**", "/img/**", "/vendor/**"};

    public static class Api {

        public static final String SYSTEM = API + Url.Page.SYSTEM;
        public static final String USER = API + Url.Page.USER;

        public static final String API_PATH_PATTERN = API + ALL_PATH_PATTERN;

        public static class System {

            public static final String API = SYSTEM + "/api";
            public static final String API_CATEGORY = SYSTEM + "/api_category";

            public static final String VIEW_PAGE = SYSTEM + "/view_page";
            public static final String VIEW_PAGE_CATEGORY = SYSTEM + "/view_page_category";

            public static final String VIEW_PAGE_API = SYSTEM + "/view_page_api";

            public static final String VIEW_PAGE_COMPONENT = SYSTEM + "/view_page_component";
            public static final String VIEW_PAGE_COMPONENT_API = SYSTEM + "/view_page_component_api";

            public static final String DICTIONARY = SYSTEM + "/dictionary";
            public static final String DICTIONARY_CATEGORY = SYSTEM + "/dictionary_category";


            public static final String OPERATION_LOG = SYSTEM + "/operation_log";

//        public static final String REQUEST_LOG = SYSTEM + "/operation_log";

            public static final String ATTACHMENT = SYSTEM + "/attachment";

            public static final String VERIFY_CODE = SYSTEM + "/verify_code";

        }

        public static class User {

            public static final String LOGIN = USER + "/login";
            public static final String REGISTER = USER + "/register";
            public static final String LOGOUT = USER + "/logout";

            public static final String FORGOT_PASSWORD = USER + "/forgot_password";

            public static final String SEND_PASSWORD_EMAIL = USER + "/send_password_email";
            public static final String SEND_PASSWORD_SMS = USER + "/send_password_sms";

            public static final String ROLE = USER + "/role";
            public static final String ROLE_AUTHORITY = USER + "/role_authority";
            public static final String USER_ROLE = USER + "/user_role";

            public static final String VIEW_MENU = USER + "/view_menu";
            public static final String VIEW_MENU_CATEGORY = USER + "/view_menu_category";

            public static final String DEPARTMENT = USER + "/department";

        }

    }

    public static class Page {

        public static final String SYSTEM = "/system";
        public static final String USER = "/user";

        public static final String INDEX = "/index";
        public static final String[] INDEX_ARRAY = {PATH_SEPARATOR, INDEX, SYSTEM, System.INDEX};

        public static class System {

            public static final String INDEX = SYSTEM + Url.Page.INDEX;

            public static final String WORKBENCH = SYSTEM + "/workbench";

            public static final String API = SYSTEM + "/api";
            public static final String API_CATEGORY = SYSTEM + "/api_category";

            public static final String DICTIONARY = SYSTEM + "/dictionary";
            public static final String DICTIONARY_CATEGORY = SYSTEM + "/dictionary_category";

            public static final String VIEW_PAGE = SYSTEM + "/view_page";
            public static final String VIEW_PAGE_CATEGORY = SYSTEM + "/view_page_category";
            public static final String VIEW_PAGE_API = SYSTEM + "/view_page_api";

            public static final String VIEW_PAGE_COMPONENT = SYSTEM + "/view_page_component";
            public static final String VIEW_PAGE_COMPONENT_API = SYSTEM + "/view_page_component_api";

            public static final String MAIL = SYSTEM + "/mail";

            public static final String OPERATION_LOG = SYSTEM + "/operation_log";

//        public static final String REQUEST_LOG = SYSTEM + "/operation_log";

            public static final String ATTACHMENT = SYSTEM + "/attachment";

        }

        public static class User {

            public static final String LOGIN_ACCOUNT_STRING = "account";
            public static final String LOGIN_PASSWORD_STRING = "password";
            public static final String LOGIN_REMEMBER_ME_STRING = "rememberMe";

            public static final String LOGIN = USER + "/login";
            public static final String LOGIN_PATH_PATTERN = LOGIN + ALL_PATH_PATTERN;

            public static final String REGISTER = USER + "/register";
            public static final String REGISTER_PATH_PATTERN = REGISTER + ALL_PATH_PATTERN;

            public static final String LOGOUT = USER + "/logout";
            public static final String LOGOUT_PATH_PATTERN = LOGIN + ALL_PATH_PATTERN;

            public static final String ROLE = USER + "/role";
            public static final String ROLE_AUTHORITY = USER + "/role_authority";

            public static final String USER_ROLE = USER + "/user_role";

            public static final String VIEW_MENU = USER + "/view_menu";
            public static final String VIEW_MENU_CATEGORY = USER + "/view_menu_category";

            public static final String DEPARTMENT = USER + "/department";


        }
    }

}
